test/application/server-commands.test.js

```diff
@@ -0,0 +1,95 @@
+const simpleGit = require("simple-git");
+const git = simpleGit();
+
+const scenarioUpToDate = {
+    modified: [],
+    ahead: 0,
+    behind: 0,
+}
+
+describe('ServerCommands', () => {
+    const scenarioModifiedOutput = {
+        modified: [
+            "package-lock.json",
+            "package.json",
+            "server_src/lib/serverCommands.js",
+        ],
+        ahead: 0,
+        behind: 0,
+    };
+
+    jest.mock("child_process", () => {
+        return {
+            exec: () => Promise.resolve()
+        }
+    });
+
+    let wasReset = null;
+    let currentScenario = scenarioUpToDate;
+    jest.mock("simple-git", () => {
+        return () => {
+            return {
+                status: () => Promise.resolve(
+                    currentScenario
```

Comment:
The last set scenario is picked up and provided as internal output of simple-git